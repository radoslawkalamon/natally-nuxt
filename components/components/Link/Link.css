.link--general {
  color: inherit;
  text-decoration: underline;
}

.link--general:hover {
  text-decoration: none;
}

.link--button {
  background-color: var(--color-accent);
  border: var(--border-size) solid var(--color-clickable-border);
  border-radius: var(--border-radius);
  color: var(--color-off-white-1);
  display: inline-block;
  font-family: var(--font-family-sans);
  font-size: var(--font-size-base);
  font-weight: bold;
  margin: 0;
  padding: 0.5em 1em;
  text-decoration: none;
  transition: opacity var(--time-animation) ease;
}

.link--button:hover {
  text-decoration: underline;
}
