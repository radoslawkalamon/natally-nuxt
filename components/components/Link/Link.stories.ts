import type { Meta, Story } from '@storybook/vue'
import ComponentsLink from '@/components/components/Link/Link.vue'

const meta: Meta = {
  title: 'Components / Link',
  component: ComponentsLink,
  argTypes: {
    type: {
      options: ['general', 'button'],
      control: {
        type: 'inline-radio'
      }
    }
  }
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: { ComponentsLink },
  props: Object.keys(argTypes),
  template: '<ComponentsLink v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  to: '/the-url/',
  type: 'general',
  label: 'Item label'
}
