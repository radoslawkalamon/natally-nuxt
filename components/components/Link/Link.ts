import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsLink',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      required: true
    },
    type: {
      type: String,
      default: 'general',
      validator: v => ['general', 'button'].includes(v)
    }
  },
  computed: {
    classNames () {
      return [
        'link',
        `link--${this.type}`
      ]
    }
  }
})
