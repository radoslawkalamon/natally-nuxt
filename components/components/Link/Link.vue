<template>
  <NuxtLink
    :class="classNames"
    v-bind="$attrs"
    v-on="$listeners"
    v-text="label"
  />
</template>

<script src="./Link.ts" lang="ts"></script>
<style src="./Link.css" scoped></style>
