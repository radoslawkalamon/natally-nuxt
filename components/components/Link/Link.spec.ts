import Link from '@/components/components/Link/Link.vue'
import { shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Components / Link', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: Link,
      description: 'General Type',
      options: {
        propsData: {
          label: 'Item label',
          to: '/test-path/',
          type: 'general'
        }
      }
    })

    shallPassUnitSanityTest({
      component: Link,
      description: 'Button Type',
      options: {
        propsData: {
          label: 'Item label',
          to: '/test-path/',
          type: 'button'
        }
      }
    })
  })
})
