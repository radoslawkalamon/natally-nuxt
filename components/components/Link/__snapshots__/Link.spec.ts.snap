// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Link Unit shall pass unit sanity test | Button Type 1`] = `"<nuxtlink-stub to=\\"/test-path/\\" class=\\"link link--button\\">Item label</nuxtlink-stub>"`;

exports[`Components / Link Unit shall pass unit sanity test | General Type 1`] = `"<nuxtlink-stub to=\\"/test-path/\\" class=\\"link link--general\\">Item label</nuxtlink-stub>"`;
